src/react/PipelinesPage.tsx

```diff
@@ -16,55 +16,74 @@
-  const [phase, setPhase] = useState('loading');
-  const [pipelines, setPipelines] = useState(null);
+import useFetch from './hooks/useFetch';
 
-  useEffect(() => {
-    const handlePipelinesResponse = (pipelineList) => {
-      const promiseList = pipelineList.map((pipelineName) =>
-        astarte.getPipelineInputConfig(pipelineName),
-      );
-      Promise.allSettled(promiseList).then((result) => {
-        const pipelineData = [];
-        result.forEach((pipelineResult) => {
-          if (pipelineResult.status === 'fulfilled') {
-            pipelineData.push(pipelineResult.value);
-          }
-        });
-        setPipelines(pipelineData);
-        setPhase('ok');
-      });
-    };
-    const handlePipelinesError = () => {
-      setPhase('err');
-    };
-    astarte.getPipelineDefinitions().then(handlePipelinesResponse).catch(handlePipelinesError);
-  }, [astarte]);
+interface NewPipelineCardProps {
+  onCreate: () => void;
+}
+
+const NewPipelineCard = ({ onCreate }: NewPipelineCardProps): React.ReactElement => (
+  <Card className="mb-4">
+    <Card.Header as="h5">New Pipeline</Card.Header>
+    <Card.Body>
+      <Card.Text>Create your custom pipeline</Card.Text>
+      <Button variant="secondary" onClick={onCreate}>
+        Create
+      </Button>
+    </Card.Body>
+  </Card>
+);
+
+interface PipelineCardProps {
+  pipeline: AstartePipeline;
+  onInstantiate: () => void;
+}
+
+const PipelineCard = ({ pipeline, onInstantiate }: PipelineCardProps): React.ReactElement => (
+  <Card className="mb-4">
+    <Card.Header as="h5">
+      <Link to={`/pipelines/${pipeline.name}`}>{pipeline.name}</Link>
+    </Card.Header>
+    <Card.Body>
+      <Card.Text>{pipeline.description}</Card.Text>
+      <Button variant="primary" onClick={onInstantiate}>
+        Instantiate
+      </Button>
+    </Card.Body>
+  </Card>
+);
+
+interface Props {
+  astarte: AstarteClient;
+  history: any;
+}
+
+export default ({ astarte, history }: Props): React.ReactElement => {
+  const pipelines = useFetch(astarte.getPipelines);
 
   let innerHTML;
 
-  switch (phase) {
+  switch (pipelines.status) {
     case 'ok':
       innerHTML = (
         <CardDeck className="mt-4">
           <NewPipelineCard onCreate={() => history.push('/pipelines/new')} />
-          {pipelines.map((pipeline, index) => (
+          {pipelines.value.map((pipeline, index) => (
             <React.Fragment key={`fragment-${index}`}>
               {index % 2 ? <div className="w-100 d-none d-md-block" /> : null}
               <PipelineCard
-                headless="true"
-                pipelineName={pipeline.name}
-                pipelineDescription={pipeline.description}
-                pipelineLongDescription={pipeline.longDescription}
-                configureCB={() => history.push(`/flows/new/${pipeline.name}`)}
-                onShow={() => history.push(`/pipelines/${pipeline.name}`)}
+                pipeline={pipeline}
+                onInstantiate={() => {
```

Comment:
if you're accepting a callback for the link "Instantiate" you can also accept a callback for the "Show" link.
Or skip the callbacks at all and define everything inside the `PipelineCard`

Reply:
Fixed